vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlCredentialsProvider.java

```diff
@@ -0,0 +1,36 @@
+package io.vertx.sqlclient;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+
+public interface SqlCredentialsProvider {
```

Comment:
From my perspective, it would be better to have `SqlCredentialsProvider` annotated with `@VertxGen`, so that Mutiny / Rx / Kotlin users have something that integrates well with the rest of their code base.

Reply:
Will do.